File: DarkOrbit 10.0/Net/netty/commands/ClientUITooltipsCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class ClientUITooltipsCommand
    {
        public static short ID = 11246;

        public List<ClientUITooltipModule> mTooltips;

        public ClientUITooltipsCommand(List<ClientUITooltipModule> pTooltips)
        {
            this.mTooltips = pTooltips;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeInt(this.mTooltips.Count);
            foreach (var c in this.mTooltips)
            {
                param1.write(c.write());
            }
            param1.writeShort(-21605);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/ClientUITooltipsCommand.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Vector;

import simulator.netty.ServerCommand;

public class ClientUITooltipsCommand
        implements ServerCommand {

    public static int ID = 4659;

    public Vector<ClientUITooltipModule> mTooltips;

    public ClientUITooltipsCommand(Vector<ClientUITooltipModule> pTooltips) {
        this.mTooltips = pTooltips;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 4;
    }

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream out) {
        try {
            out.writeInt(this.mTooltips.size());
            for (ClientUITooltipModule c : this.mTooltips) {
                c.write(out);
            }
            out.writeShort(9452);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}